code/weapon/weapons.cpp

```diff
@@ -7622,21 +7622,21 @@ void weapon_hit( object * weapon_obj, object * other_obj, vec3d * hitpos, int qu
 		hit_angle = vm_vec_delta_ang(hitnormal, &reverse_incoming, nullptr);
 	}
 
-	if (wip->conditional_impacts.size() > 0 && other_obj != nullptr && (other_obj->type == OBJ_SHIP || other_obj->type == OBJ_WEAPON)) {
-		if (other_obj->type == OBJ_SHIP) {
-			shipp = &Ships[other_obj->instance];
+	if (!wip->conditional_impacts.empty() && impacted_obj != nullptr && (impacted_obj->type == OBJ_SHIP || impacted_obj->type == OBJ_WEAPON)) { // NOLINT(readability-simplify-boolean-expr)
```

Comment:
This `// NOLINT` should be removed because this particular clang alert wasn't for a boolean expression.

Reply:
ah that's right, just pushed, thanks!